synapse/tests/test_runtime.py

```diff
@@ -80,7 +80,8 @@ def runStormBasics(self, core):
         self.sorteq(core.eval('inet:ipv4="127.0.0.1" inet:ipv4->inet:dns:a:ipv4 inet:dns:a:fqdn->inet:fqdn'), [t6, t7])
 
         # test join operator basics
-        self.sorteq(core.eval('inet:ipv4="127.0.0.1" join(inet:ipv4:cc)'), [t0, t1, t2])
+        self.raises(BadSyntaxError, core.eval, 'inet:ipv4="127.0.0.1" join(inet:ipv4:cc)')
```

Comment:
Users must now specify a dst and src

Reply:
Looked at your comments in chat...you said "under the hood, if you don't provide 'src' it makes 'src' and 'dst' the same". Comparing join() to pivot() (and we've described join() as "a pivot() that retains the original working set"), pivot() takes params in the order 'src', 'dst', and 'src' is optional - if it's not provided, it defaults to the primary property of the working set of nodes.

It seems like join() should behave the same way:
1. take params in same order as join( 'src' , 'dst' ) (US-1512)
2. if no 'src' provided, assume the primary prop of the working set (see https://vertexprojectsynapse.readthedocs.io/en/latest/synapse/userguides/ug015_storm_ref_pivot.html#pivot)

Thoughts?

Reply:
Lol, should have read this first. :D Just gave the same feedback above. :)

Reply:
Yep, I'm already working on that and can just add it to this PR.

Reply:
I just threw this up first because I think it can be merged independently of how we make join work like pivot.